beauty-consulting-app/app/assets/components/EditProfileViewModal.js

```diff
@@ -0,0 +1,199 @@
+import DateTimePicker from "@react-native-community/datetimepicker";
+import { UserContext, updateUserContext } from "../../contexts/userContext";
+import handleHTTPError from "utils/errorHandling";
+import api from "utils/axios";
+
+const EditProfileViewModal = ({ visible, onClose, profileDetails }) => {
+	const [formData, setFormData] = useState({
+		birthday: new Date(), // default to current date
+		gender: "",
+		phoneNumber: "",
+		email: "",
+	});
+
+	const userContext = useContext(UserContext);
+
+	useEffect(() => {
+		if (profileDetails) {
+			setFormData({
+				birthday: new Date(profileDetails.birthday),
+				gender: profileDetails.gender,
+				phoneNumber: profileDetails.phoneNumber,
+				email: profileDetails.email,
+			});
+		}
+	}, [profileDetails]);
+
+	const handleInputChange = (field, value) => {
+		setFormData((prev) => ({
+			...prev,
+			[field]: value,
+		}));
+	};
+
+	const handleDateChange = (event, selectedDate) => {
+		if (event.type === "set") {
+			const currentDate = selectedDate || formData.birthday;
+			setFormData((prev) => ({
+				...prev,
+				birthday: currentDate,
+			}));
+		}
+	};
+
+	const saveAndClose = async () => {
+		updatedData = {
+			...userContext,
+			email: formData.email,
+			info: {
+				phoneNumber: formData.phoneNumber,
+				gender: formData.gender,
+				birthday: formData.birthday,
+			},
+		};
+		userContext.updateUserContext(updatedData);
+		try {
+			await api.put(`client/${userContext.username}`, updatedData);
+		} catch (error) {
+			handleHTTPError(error);
+		}
+		onClose(formData);
+	};
+
+	return (
+		<Modal visible={visible} transparent={false} animationType="slide">
+			<View style={styles.headerRow}>
+				<Text style={styles.title}>Edit Details</Text>
+				<TouchableOpacity
+					style={globalStyles.button}
+					onPress={saveAndClose}
+				>
+					<Text style={globalStyles.buttonText}>Save Edits</Text>
+				</TouchableOpacity>
+			</View>
+			<View style={styles.container}>
+				<View style={styles.infoRow}>
+					<AgeIcon width={20} height={20} style={styles.icon} />
+					<Text style={styles.label}>Birthday:</Text>
+					<DateTimePicker
+						value={formData.birthday}
+						mode="date"
+						onChange={handleDateChange}
+					/>
+				</View>
+
+				<View style={styles.infoRow}>
+					<PhoneIcon width={20} height={20} style={styles.icon} />
+					<Text style={styles.label}>Phone:</Text>
+					<TextInput
+						style={[styles.input, styles.value]}
+						defaultValue={formData.phoneNumber}
+						keyboardType="phone-pad"
+						onChangeText={(value) =>
+							handleInputChange("phoneNumber", value)
+						}
+					/>
+				</View>
+
+				<View style={styles.infoRow}>
+					<GenderIcon width={20} height={20} style={styles.icon} />
+					<Text style={styles.label}>Gender:</Text>
+					<TextInput
+						style={[styles.input, styles.value]}
+						value={formData.gender}
+						onChangeText={(value) =>
+							handleInputChange("gender", value)
+						}
+					/>
```

Comment:
Can this have only the selection choices we use during signup?

Reply:
yes